From d248840337ceae97da54c77a919cf46be8817c53 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 30 Aug 2022 15:58:32 -0700
Subject: [PATCH] Add needed headers for including mbstate_t and exit()

Upstream-Status: Inappropriate [Reconfigure will solve it]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index a49b1759..d98f7019 100755
--- a/configure
+++ b/configure
@@ -3762,6 +3762,7 @@ rm -f "conftest.$ac_objext" "conftest.$ac_ext"
   cat >"conftest.$ac_ext" <<_ACEOF
 #line 3763 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 $ac_declaration
 int
 main (void)
@@ -13869,6 +13870,7 @@ cat >"conftest.$ac_ext" <<_ACEOF
 #include <stdlib.h>
 #include <stdarg.h>
 #include <stdio.h>
+#include <wchar.h>
 #ifdef HAVE_LIBUTF8_H
 #include <libutf8.h>
 #endif
