Upstream-Status: Backport

diff -durpN dbus-glib-0.112/dbus/dbus-gvalue.c dbus-glib-0.114/dbus/dbus-gvalue.c
--- dbus-glib-0.112/dbus/dbus-gvalue.c	2021-03-26 13:40:43.000000000 +0100
+++ dbus-glib-0.114/dbus/dbus-gvalue.c	2025-03-13 15:24:40.000000000 +0100
@@ -545,11 +555,11 @@ demarshal_basic (DBusGValueMarshalCtx
     {
     case DBUS_TYPE_BOOLEAN:
       {
-	dbus_bool_t bool;
+	dbus_bool_t b;
         if (!G_VALUE_HOLDS (value, G_TYPE_BOOLEAN))
           goto invalid_type;
-	dbus_message_iter_get_basic (iter, &bool);
-	g_value_set_boolean (value, bool);
+	dbus_message_iter_get_basic (iter, &b);
+	g_value_set_boolean (value, b);
 	return TRUE;
       }
     case DBUS_TYPE_BYTE:
